extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -494,20 +505,28 @@ public RunnerTaskState getRunnerTaskState(String taskId)
     }
   }
 
-  public class K8sWorkItem extends TaskRunnerWorkItem
+  public static class K8sWorkItem extends TaskRunnerWorkItem
   {
     private final Task task;
+    private KubernetesPeonClient client;
 
-    private K8sWorkItem(Task task, ListenableFuture<TaskStatus> statusFuture)
+    public K8sWorkItem(KubernetesPeonClient client, Task task, ListenableFuture<TaskStatus> statusFuture)
     {
       super(task.getId(), statusFuture);
       this.task = task;
+      this.client = client;
     }
 
-    private K8sWorkItem(Task task, ListenableFuture<TaskStatus> statusFuture, DateTime createdTime)
+    public K8sWorkItem(
+        KubernetesPeonClient client,
+        Task task,
+        ListenableFuture<TaskStatus> statusFuture,
+        DateTime createdTime
+    )
     {
       super(task.getId(), statusFuture, createdTime, createdTime);
       this.task = task;
+      this.client = client;
```

Comment:
where does it get accessed?

Reply:
Used for testing.  I needed to inject the client to mock out